src/client/Microsoft.Identity.Client.Brokers/WamAdapters.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.Utils;
+using static Microsoft.Identity.Client.Broker.RuntimeBroker;
+
+namespace Microsoft.Identity.Client.Broker
+{
+    internal class WamAdapters
+    {
+        private const string WamErrorPrefix = "WAM Error ";
+        
+        //MSA-PT Auth Params
+        private const string NativeInteropMsalRequestType = "msal_request_type";
+        private const string ConsumersPassthroughRequest = "consumer_passthrough";
+
+        //MSAL Runtime Error Response 
+        private enum ResponseStatus : int
+        {
+            Unexpected = 0,
+            Reserved = 1,
+            InteractionRequired = 2,
+            NoNetwork = 3,
+            NetworkTemporarilyUnavailable = 4,
+            ServerTemporarilyUnavailable = 5,
+            ApiContractViolation = 6,
+            UserCanceled = 7,
+            ApplicationCanceled = 8,
+            IncorrectConfiguration = 9,
+            InsufficientBuffer = 10,
+            AuthorityUntrusted = 11,
+            UserSwitch = 12,
+            AccountUnusable = 13,
+            UserDataRemovalRequired = 14
+        };
+
+        /// <summary>
+        /// Create WAM Error Response
+        /// </summary>
+        /// <param name="authResult"></param>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalClientException"></exception>
+        /// <exception cref="MsalUiRequiredException"></exception>
+        /// <exception cref="MsalServiceException"></exception>
+        internal static MsalException CreateWamErrorResponse(
```

Comment:
This method throws the exception everytime, it does not return it.

Reply:
void